service-discovery/stork-consul/src/test/java/io/quarkus/ts/stork/StorkServiceDiscoveryIT.java

```diff
@@ -0,0 +1,162 @@
+            .withProperty("stork.pong.service-discovery.consul-host", () -> getConsultEndpoint(consul.getConsulEndpoint()));
+
+    @Test
+    @Order(1)
+    public void invokeServiceByName() {
+        String response = makePxngCall("pung");
+        assertThat("Service discovery by name fail.", PREFIX + "pung", is(response));
+    }
+
+    @Test
+    @Order(2)
+    public void storkLoadBalancerByRoundRobin() {
+        Map<String, Integer> uniqueResp = new HashMap<>();
+        final int requestAmount = 100;
+        final int roundRobinError = (requestAmount / 2) - 1;
+        for (int i = 0; i < requestAmount; i++) {
+            String response = makePxngCall("pong");
+            if (uniqueResp.containsKey(response)) {
+                uniqueResp.put(response, uniqueResp.get(response) + 1);
+            } else {
+                uniqueResp.put(response, 1);
+            }
+        }
+
+        assertThat("Only 2 services should response", uniqueResp.size(), is(2));
+        assertThat("Unexpected service names", uniqueResp.keySet(),
+                hasItems(PREFIX + DEFAULT_PONG_RESPONSE, PREFIX + DEFAULT_PONG_REPLICA_RESPONSE));
+        assertThat("Load balancer doesn't behaves as round-robin", uniqueResp.get(PREFIX + DEFAULT_PONG_RESPONSE),
+                is(greaterThanOrEqualTo(roundRobinError)));
+        assertThat("Load balancer doesn't behaves as round-robin", uniqueResp.get(PREFIX + DEFAULT_PONG_REPLICA_RESPONSE),
+                is(greaterThanOrEqualTo(roundRobinError)));
+    }
+
+    @Test
+    @Order(3) // run this test always last, because we are dropping pongService
+    @Disabled("Looks that service-discovery.refresh-period is not aware about service readiness")
+    //TODO https://github.com/quarkusio/quarkus/issues/24343
+    public void storkLoadBalancerServiceNodeDown() {
+        final int requestAmount = 100;
+        final int nodeDownThreshold = 50;
+        final Duration nodeDownRecoveryTimeout = Duration.ofSeconds(10);
+        for (int i = 0; i < requestAmount; i++) {
+            Awaitility.await().atMost(nodeDownRecoveryTimeout).untilAsserted(() -> makePxngCall("pong"));
+            if (i > nodeDownThreshold) {
+                pongService.stop();
+            }
+        }
+
+        String respBody = makePxngCall("pong");
+        assertThat("Unexpected service name", respBody, is(PREFIX + DEFAULT_PONG_REPLICA_RESPONSE));
+    }
+
+    private String makePxngCall(String pxngPath) {
+        return mainPingService
+                .given()
+                .get("/ping/" + pxngPath).then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().body().asString();
+    }
+
+    private static String getConsultEndpoint(String endpoint) {
+        return endpoint.replaceFirst(":\\d+", "");
+    }
+
+    private static String getAvailablePort() {
+        int port = 0;
+        try (ServerSocket serverSocket = new ServerSocket(port)) {
+            port = serverSocket.getLocalPort();
```

Comment:
Return condition can be moved here(unless you want to return zero port, of course :)